AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProviderResponse.swift

```diff
@@ -66,6 +68,33 @@ extension RealtimeConnectionProviderResponse: Decodable {
 /// https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html#error-message
 extension RealtimeConnectionProviderResponse {
 
+    func toConnectionProviderError(connectionState: ConnectionState) -> ConnectionProviderError {
+        // If it is Unauthorized, return `.other` error.
+        guard !isUnauthorizationError() else {
+            return .other(errorDescription: "Unauthorized", error: nil, payload: payload)
```

Comment:
it seems that here and at line 91 we use the `.other` error for two different scenarios, which I believe exposes a problem with this kind of general error, like "other". I would lean towards using `.unauthorized` and `.unknown` to replace these two occurrences with something that better communicates what's happening here to the consumers of this API.

Reply:
i remember thinking of adding `.unauthorized` but now i can't remember the reasoning for why we did not want to at that time for making a breaking change to consumers. Considering this isn't a public facing package, my thoughts right now are leaning towards making the correct design decisions and version accordingly